libs/payments/customer/src/lib/invoice.manager.ts

```diff
@@ -43,12 +43,32 @@ export class InvoiceManager {
     private currencyManager: CurrencyManager
   ) {}
 
+  // Finalize an invoice, rejecting re-finalization of an invoice
   async finalizeWithoutAutoAdvance(invoiceId: string) {
     return this.stripeClient.invoicesFinalizeInvoice(invoiceId, {
       auto_advance: false,
     });
   }
 
+  // Finalize an invoice, permitting re-finalization of an invoice
+  async safeFinalizeWithoutAutoAdvance(invoiceId: string) {
+    try {
+      return await this.stripeClient.invoicesFinalizeInvoice(invoiceId, {
+        auto_advance: false,
+      });
+    } catch (err) {
+      // This is Stripe's only unique way of identifying this error. Remove as part of FXA-11460
+      if (
+        err?.raw?.message !==
+        "This invoice is already finalized, you can't re-finalize a non-draft invoice."
+      ) {
+        throw err;
+      } else {
+        return await this.stripeClient.invoicesRetrieve(invoiceId);
```

Comment:
nit. no need to `await` here.

Reply:
nice. Its needed on the `return await...` call within the `try` block, but not here